test-requirements.txt

```diff
@@ -15,3 +15,7 @@ py>=1.5.2
 virtualenv<20
 setuptools!=50
 importlib-metadata==0.20
+# stubgen compatibility with pybind
+pybind11
+numpy
+git+https://github.com/sizmailov/python_example.git@mypy-stubgen-test
```

Comment:
I've took https://github.com/pybind/python_example as a starting point and added a number of peculiar examples to test capabilities of stubgen (spoiler: currently it's quite far from perfect, I'll submit more changes in next PRs).

I'm not sure if it's appropriate to have a git-requirement with link to private repo. The dependency is small and probably it's fine to copy it as whole (just needed parts) somewhere in mypy.